lib/widgets.php

```diff
@@ -213,3 +213,23 @@ function gutenberg_create_wp_area_post_type() {
 add_action( 'init', 'gutenberg_create_wp_area_post_type' );
 
 add_filter( 'sidebars_widgets', 'Experimental_WP_Widget_Blocks_Manager::swap_out_sidebars_blocks_for_block_widgets' );
+
```

Comment:
I guess we can move this logic to a customizer.php file. 
For reference, this code was added because of the global widgets global used in this function:
```
	public function sanitize_sidebar_widgets_js_instance( $widget_ids ) {
		global $wp_registered_widgets;
		$widget_ids = array_values( array_intersect( $widget_ids, array_keys( $wp_registered_widgets ) ) );
		return $widget_ids;
	}
```
Does not contains the old format but contains the new format, I guess the best solution is to undestand why the global does not contain the backcompatible format and fix that problem as it may cause problems with plugins. If that is not possible I guess we can use this filter but instead of unsetting sanitize_js_callback I think we can make it point to new function where we compute the correct inputs based on the post id and we pass this values to the function that was set in $args['sanitize_js_callback'] before.
This does not seem a problem form this PR so I think we can address it separately provided we have a "Todo:" note to not miss this task.

Reply:
Yeah I wrote that in the documentation:

https://github.com/WordPress/gutenberg/blob/4899f9a73eef983e7d26f0b7a29a7a4688b02e24/lib/widgets.php#L218-L221

But, the issue is that `$widget_ids` is an int when it should be an array.

This filter, https://github.com/WordPress/gutenberg/blob/4899f9a73eef983e7d26f0b7a29a7a4688b02e24/lib/class-experimental-wp-widget-blocks-manager.php#L341, fixes that, but it doesn't run for that part of core.

I've moved it to customizer.php and added a TODO.